### PR TITLE
Add Evm Contracts.

### PR DESCRIPTION
**Pull request by Sriep**
New EvmContract for deploying and calling methods on a EVM contract.

### CODE REVIEW

File: client/evm_contract.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		return nil
+	}
+	return &EvmContract{
+		client:  client,
+		Address: loom.UnmarshalAddressPB(response.Contract),
+		Name:    name,
+	}
+}
+
+func (c *EvmContract) CallEvm(input []byte, signer auth.Signer) ([]byte, error) {
```
Comment (enlight) [inline client/evm_contract.go]: `CallEvm` -> `Call`

File: client/evm_contract.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	callerAddr := loom.Address{
+		ChainID: c.client.GetChainID(),
+		Local:   loom.LocalAddressFromPublicKey(signer.PublicKey()),
+	}
+	return c.client.CommitCallTx(callerAddr, c.Address, signer, vm.VMType_EVM, input)
+}
+
+func (c *EvmContract) StaticCallEvm(input []byte) ([]byte, error) {
```
Comment (enlight) [inline client/evm_contract.go]: `StaticCallEvm` -> `StaticCall`

File: client/evm_contract.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	return &EvmContract{
+		client: client,
+		Address: loom.Address{
+			ChainID: client.GetChainID(),
+			Local:   contractAddr,
+		},
+	}
+}
+
+func NewDeployEvmContract(client *DAppChainRPCClient, signer auth.Signer, byteCode []byte, name string) *EvmContract {
```
Comment (enlight) [inline client/evm_contract.go]: `NewDeployEvmContract` -> `DeployContract`... and don't swallow the errors, return them.

File: client/evm_contract.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+// Contract provides a thin abstraction over DAppChainClient that makes it easier to perform
+// read & write operations on a contract running on a Loom DAppChain.
+type EvmContract struct {
+	client  *DAppChainRPCClient
+	Address loom.Address
+	Name    string
+}
+
+func NewEvmContract(client *DAppChainRPCClient, contractAddr loom.LocalAddress) *EvmContract {
```
Comment (enlight) [inline client/evm_contract.go]: `NewEvmContract` -> `NewContract`

Comment (Sriep) [reply client/evm_contract.go]: This conflicts with the `NewContract` in contract.go. Unless evm_contract is in a separate package.

**Pull request merged**